test/e2e/cypress/integration/sap_system_details.js

```diff
@@ -0,0 +1,138 @@
+  });
+
+  beforeEach(() => {
+    cy.Cookies.preserveOnce(sessionCookie);
+  });
+
+  describe('SAP system details page is available', () => {
+    it(`should display the "${selectedSystem.Sid}" system details page`, () => {
+      cy.get('h1').should('contain', 'SAP System Details');
+      cy.get('div')
+        .contains('Name')
+        .next()
+        .should('contain', selectedSystem.Sid);
+      cy.get('div')
+        .contains('Type')
+        .next()
+        .should('contain', selectedSystem.Type);
+    });
+
+    it(`should display "Not found" page when SAP system doesn't exist`, () => {
+      cy.visit(`/sap_systems/other`, { failOnStatusCode: false });
+      cy.url().should('include', `/sap_systems/other`);
+      cy.get('div').should('contain', 'Not Found');
+    });
+  });
+
+  describe('The system layout shows all the running instances', () => {
+    before(() => {
+      cy.visit(`/sap_systems/${selectedSystem.Id}`);
+      cy.url().should('include', `/sap_systems/${selectedSystem.Id}`);
+    });
+
+    selectedSystem.Hosts.forEach((instance, index) => {
+      it(`should show hostname "${instance.Hostname}" with the correct values`, () => {
+        cy.get('table.table-fixed')
+          .eq(0)
+          .find('tr')
+          .eq(index + 1)
+          .find('td')
+          .as('tableCell');
+        cy.get('@tableCell').eq(0).should('contain', instance.Hostname);
+        cy.get('@tableCell').eq(1).should('contain', instance.Instance);
+        instance.Features.split('|').forEach((feature) => {
+          cy.get('@tableCell').eq(2).should('contain', feature);
+        });
+        cy.get('@tableCell').eq(3).should('contain', instance.HttpPort);
+        cy.get('@tableCell').eq(4).should('contain', instance.HttpsPort);
+        cy.get('@tableCell').eq(5).should('contain', instance.StartPriority);
+        cy.get('@tableCell').eq(6).should('contain', instance.Status);
+        cy.get('@tableCell')
+          .eq(6)
+          .find('span')
+          .should('have.class', healthMap[instance.StatusBadge]);
+      });
+    });
+
+    Object.entries(healthMap).forEach(([state, health]) => {
+      it(`should show ${state} badge in instace when SAPControl-${state} state is received`, () => {
+        cy.loadScenario(`sap-system-detail-${state}`);
+        // using row 3 as the changed instance is the 3rd in order based on instance_number
+        cy.get('table.table-fixed')
+          .eq(0)
+          .find('tr')
+          .eq(3)
+          .find('td')
+          .as('tableCell');
+        cy.get('@tableCell').eq(6).should('contain', `SAPControl-${state}`);
+        cy.get('@tableCell').eq(6).find('span').should('have.class', health);
+      });
+    });
+    /* This test is commented because there is not any option to remove added SAP instances or
+    resetting the database afterwards, and it affects the rest of the test suite.
+    it(`should show a new instance when an event with a new SAP instance is received`, () => {
```

Comment:
The test actually works, and the functionality too, but as future tests are only based in the original 27 nodes landscape, and we are not able to remove the recently added instance, everything else starts failing.
Here there are 2 solutions:
- We implement the option to reset the database and start in a clean env
- We implement the SAP instances de-registration (this is actually a feature that is already in our backlog)

Reply:
Implementation of the SAP instance de-registration is, as you mention, something on the backlog we'll be working on.
So I'd keep it out from this discussion.

If I am understanding properly, the problem is that when we load the scenario `sap-system-detail-NEW` - or any scenario - we change the state and other tests would be negatively impacted by this.
```js
cy.loadScenario(`sap-system-detail-NEW`);
```
By having the `resetDatabase` command in cypress we'd be able to clean up the state and restart.
I am afraid this might turn out not that straightforward because elixir complains about playing around with the database it is connected to (dropping, recreating databases/tables)

There might be other strategies playing with also photofinish but I am not sure right now.
We might want to further discuss this.

I believe since we are working on the e2e debt, it would be beneficial to also take into account this activity.

Reply:
@nelsonkopliku Ticket for this `reset database` topic created in our team backlog as `tech-debt`